lib/Driver/ToolChains.h

```diff
@@ -96,12 +96,20 @@ class LLVM_LIBRARY_VISIBILITY GenericUnix : public ToolChain {
   ~GenericUnix() = default;
 };
 
+class LLVM_LIBRARY_VISIBILITY Android : public GenericUnix {
+protected:
+  std::string getTargetForLinker() const override;
+
+  bool shouldProvideRPathToLinker() const override;
+public:
+  Android(const Driver &D, const llvm::Triple &Triple) : GenericUnix(D, Triple) {}
+  ~Android() = default;
+};
+
 class LLVM_LIBRARY_VISIBILITY Cygwin : public GenericUnix {
 protected:
   std::string getDefaultLinker() const override;
 
-  bool shouldSpecifyTargetTripleToLinker() const override;
```

Comment:
Neat trick! I'll add an `ANDROID-armv7-NEGATIVE-NOT: -Xlinker -rpath`. However I think the Cygwin negative expectation won't work here, since we explicitly invoke `%swiftc_driver -target x86_64-unknown-windows-cygnus` in the first place. So `-target` does appear in the output...